### ISSUE TITLE
Build Errors on MacOS arm64 in source build of UE5.3.2

### ISSUE THREAD

**Issue by JupiterJones** [slowburn-dev/DataConfig#6]
```
/Users/jupiter/TestProject/Plugins/DataConfig/Intermediate/Build/Mac/UnrealEditor/Inc/DataConfigCore/UHT/DcTypes.gen.cpp:152:3: error: no matching function for call to 'UEArrayCountHelper'
                UE_ARRAY_COUNT(Z_Construct_UEnum_DataConfigCore_EDcDataEntry_Statics::Enumerators),
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jupiter/UnrealEngine/UE_5.3.2_Source/Engine/Source/Runtime/Core/Public/Templates/UnrealTemplate.h:197:41: note: expanded from macro 'UE_ARRAY_COUNT'
#define UE_ARRAY_COUNT( array ) (sizeof(UEArrayCountHelper(array)) - 1)
                                        ^~~~~~~~~~~~~~~~~~
/Users/jupiter/UnrealEngine/UE_5.3.2_Source/Engine/Source/Runtime/Core/Public/Templates/UnrealTemplate.h:190:7: note: candidate template ignored: substitution failure [with T = const UECodeGen_Private::FEnumeratorParam[]]: invalid application of 'sizeof' to an incomplete type 'const UECodeGen_Private::FEnumeratorParam[]'
        auto UEArrayCountHelper(T& t) -> typename TEnableIf<__is_array(T), char(&)[sizeof(t) / sizeof(t[0]) + 1]>::Type;
             ^                                                                           ~
```
There are also "expected unqualified-id" errors wherever `Nil` appears. eg.
```
case EDcDataEntry::Nil:
if (Next == EDcDataEntry::Nil)
FDcDataVariant() : DataType(EDcDataEntry::Nil) {}
DataType = EDcDataEntry::Nil;
enum class EDcDataEntry : uint16 { Nil, ...
```
and many more.

Is there something I've missed?

apart from those 2 issue - it all looks good up to this point :)

**Comment by jagt** [comment]
Hi, thanks for reporting. Can you attach the full log so I can take a look?
I don't have a mac at hand so I'll try cross compiling to see if it's possible to reproduce.

Also are you building other targets like Win64? Is there any issue?

**Comment by JupiterJones** [comment]
I have a similar issue - No intel Win64 so I can't test that sorry.
Let me know if you need any more information, or if there's anything I can do to help.

[DataConfigBuildLog.txt](https://github.com/slowburn-dev/DataConfig/files/14325460/DataConfigBuildLog.txt)

**Comment by jagt** [comment]
From the looks of it seems OSX builds includes `objc.h` which has a define for `Nil`, which is conflicting with `EDcDataEntry::Nil`.

This is an unfortunate coincidence 🤦 I'll try do a version with `EDcDataEntry::Nil` renamed to something else but it might take a while.

**Comment by JupiterJones** [comment]
Hmmm yes that is unfortunate. It's the perfect name already. coming from a smalltalk background, I'd suggest "nil" but I know that isn't the fashion for c++ :)

**Comment by jagt** [comment]
@JupiterJones I've got a new release here: <https://github.com/slowburn-dev/DataConfig/releases/download/1.5.0_pre/DataConfig_1.5.0_UE5_pre.zip>. Note that this git repo isn't updated yet, you'll need to get the zip directly.

Can you try this one and see if it works on OSX? If it works I'll get 1.5.0 out soon with this change. Thanks a ton!

**Comment by JupiterJones** [comment]
[DataConfigBuildLog20022024.txt](https://github.com/slowburn-dev/DataConfig/files/14335813/DataConfigBuildLog20022024.txt)

None solved the problem! Nice choice.

As with all great adventures, the compiler likes to reveal the mysteries in layers :) 4 little type constants for msgpack (_UINT8 - 64) feel too qualified for their job.

Attached is the build log.

**Comment by jagt** [comment]
Hey thanks for the feedback. It's likely `_UINT8` are also defined somewhere. I'll get these renamed soon.

If you have time can you try locally rename these to something like `_UINT8_MSGPACK` to see if there're other issues?

I might really need to get a mac to test these stuffs but it's not happening that fast :)

**Comment by JupiterJones** [comment]
Yes it builds without error. Also, more than happy to test builds in the mean time. :)

**Comment by jagt** [comment]
I've just published [1.5.0](https://github.com/slowburn-dev/DataConfig/releases/tag/1.5.0).

Can you try if this works? If that's the case I'll get this issue closed :)

**Comment by JupiterJones** [comment]
Looks good to me. Consider it closed. Thanks for the quick turnaround.
